test/pipeline/integration/test_fixed_merging_pipeline.py

```diff
@@ -13,12 +13,10 @@
 
 
 def test_pipeline(tmp_path):
-    add_two = AddFixedValue(add=2)
-
     pipeline = Pipeline()
-    pipeline.add_component("first_addition", add_two)
-    pipeline.add_component("second_addition", add_two)
-    pipeline.add_component("third_addition", add_two)
+    pipeline.add_component("first_addition", AddFixedValue(add=2))
+    pipeline.add_component("second_addition", AddFixedValue(add=2))
+    pipeline.add_component("third_addition", AddFixedValue(add=2))
```

Comment:
Adding the same instance ad different components had side effects with the dictionary implementation. I think we should just not support this use case so I'm removing it from the tests